{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1aff004e",
   "metadata": {},
   "source": [
    "# SpaCy Crosslingual Coreference\n",
    "The notebook checks whether the crosslingual coreference model can be used to reliably resolve coreferences in articles of the extracted dataset.\n",
    "\n",
    "Source: https://github.com/Pandora-Intelligence/crosslingual-coreference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cb65fab2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package omw-1.4 to /home/florian/nltk_data...\n",
      "[nltk_data]   Package omw-1.4 is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import scispacy\n",
    "import spacy\n",
    "import crosslingual_coreference\n",
    "from crosslingual_coreference import Predictor\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dc8b2dd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# reads csv as dataset and stores the column 'fulltext' in a list, removes duplicates\n",
    "df = pd.read_csv('cleaned_fulltext_articles.csv')\n",
    "list_text = df['fulltext'].dropna().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "896f500d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some weights of the model checkpoint at nreimers/mMiniLMv2-L12-H384-distilled-from-XLMR-Large were not used when initializing XLMRobertaModel: ['lm_head.layer_norm.weight', 'lm_head.layer_norm.bias', 'lm_head.dense.weight', 'lm_head.dense.bias', 'lm_head.bias']\n",
      "- This IS expected if you are initializing XLMRobertaModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).\n",
      "- This IS NOT expected if you are initializing XLMRobertaModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n",
      "Some weights of XLMRobertaModel were not initialized from the model checkpoint at nreimers/mMiniLMv2-L12-H384-distilled-from-XLMR-Large and are newly initialized: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<crosslingual_coreference.CrossLingualPredictorSpacy.CrossLingualPredictorSpacy at 0x7fbf8fbf9d90>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# loads spacy model and adds coreference mode to the pipeline\n",
    "nlp = spacy.load('en_core_web_sm')\n",
    "nlp.add_pipe(\n",
    "    'xx_coref', config={'chunk_size': 2500, 'chunk_overlap': 2, 'device': -1}) # builds to not overload device = cpu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "292ac52c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/florian/python_scripts/test/lib/python3.8/site-packages/allennlp/modules/token_embedders/pretrained_transformer_embedder.py:385: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').\n",
      "  num_effective_segments = (seq_lengths + self._max_length - 1) // self._max_length\n"
     ]
    }
   ],
   "source": [
    "# processes an article with nlp to doc object\n",
    "doc = nlp(list_text[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d5709de8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[[34, 35], [83, 84], [281, 282]], [[50, 50], [399, 400]], [[106, 112], [121, 121]], [[108, 110], [131, 133], [163, 165]], [[108, 112], [244, 248]], [[112, 112], [248, 248]], [[234, 235], [241, 248], [260, 260]], [[293, 296], [309, 332], [347, 352], [364, 367], [377, 383]], [[293, 296], [309, 332], [364, 367], [377, 383]], [[468, 486], [485, 485]], [[501, 501], [572, 572]], [[719, 721], [771, 772]], [[795, 797], [806, 808]], [[806, 811], [1107, 1114], [1122, 1129], [1136, 1136], [1141, 1141]], [[806, 811], [1107, 1114], [1122, 1129], [1136, 1136], [1141, 1141], [1188, 1196], [1333, 1341], [1352, 1360]], [[810, 811], [1093, 1096], [1111, 1114], [1126, 1129]], [[810, 811], [1093, 1096], [1111, 1114], [1126, 1129], [1162, 1166], [1192, 1196], [1337, 1341], [1356, 1360], [1375, 1378], [1380, 1380], [1420, 1424]], [[868, 873], [905, 910], [950, 951], [970, 975], [1002, 1007]], [[871, 873], [897, 899], [908, 910], [973, 975], [1005, 1007], [1107, 1109], [1122, 1124]], [[871, 873], [897, 899], [908, 910], [973, 975], [1005, 1007], [1107, 1109], [1122, 1124], [1175, 1177], [1188, 1190], [1206, 1208], [1333, 1335], [1352, 1354], [1461, 1463], [1489, 1491], [1500, 1502]], [[871, 873], [897, 899], [908, 910], [973, 975], [1005, 1007], [1107, 1109], [1122, 1124], [1175, 1177], [1188, 1190], [1206, 1208], [1333, 1335], [1352, 1354], [1461, 1463], [1489, 1491], [1500, 1502], [1536, 1538], [1565, 1567], [1642, 1644], [1716, 1718], [1735, 1737], [1775, 1777]], [[1099, 1099], [1171, 1171]], [[1198, 1198], [1370, 1370]], [[1348, 1360], [1391, 1393]], [[1349, 1349], [1392, 1392]], [[1458, 1463], [1497, 1502]], [[1458, 1463], [1497, 1502], [1533, 1538], [1562, 1567], [1617, 1618]], [[1557, 1560], [1706, 1707]], [[1624, 1632], [1716, 1724], [1735, 1743]], [[1628, 1632], [1720, 1724], [1739, 1743], [1779, 1783]], [[1630, 1630], [1722, 1722]], [[1929, 1929], [1939, 1939]], [[1929, 1930], [1939, 1940]], [[2099, 2101], [2128, 2130]], [[2099, 2104], [2128, 2133]], [[2103, 2104], [2132, 2133]], [[2220, 2222], [2229, 2229], [2318, 2320]], [[2328, 2330], [2338, 2338]], [[2415, 2420], [2502, 2503]], [[2872, 2874], [2925, 2927]], [[2876, 2880], [2947, 2951]], [[3122, 3124], [3130, 3136], [3196, 3197]], [[3268, 3272], [3417, 3423]], [[3803, 3808], [3803, 3809], [3834, 3836]], [[3803, 3808], [3834, 3836]], [[3892, 3894], [4033, 4035], [4061, 4063], [4152, 4154], [4181, 4183], [4315, 4317]], [[3913, 3917], [4037, 4041], [4200, 4204], [4319, 4323]], [[3913, 3917], [4037, 4041], [4200, 4204], [4319, 4323], [4432, 4436]], [[3999, 3999], [4242, 4242]], [[4033, 4041], [4315, 4323]], [[4033, 4041], [4315, 4323], [4428, 4436]], [[4045, 4047], [4164, 4166], [4327, 4329]], [[4202, 4202], [4321, 4321]], [[4344, 4346], [4428, 4430]]]\n"
     ]
    }
   ],
   "source": [
    "print(doc._.coref_clusters)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "598f5c61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ANI, average nucleotide identity; dDDH, digital DNA–DNA hybridization; KEGG, Kyoto Encyclopedia of Genes and Genomes; R2A, Reasoner's 2A. The newly sequenced data included in this work are deposited under the nucleotide accession numbers: MZ919349 and MZ920050 and under the Bioproject accession numbers JAIQDI000000000, JAJNEC000000000 and JAIQDJ000000000 at a public domain server in the National Center for Biotechnology Information database. All supporting data, code and protocols have been provided within the article or through supplementary data files. A supplementary table and further supplementary files can be found at: https://doi.org/10.6084/m9.figshare.18220907.v1. Although members of the genera Niabella and Thermomonas are often isolated from similar environmental samples, they belong to the Bacteroidetes and Proteobacteria, respectively. The genus Niabella is a member of the family Chitinophagaceae. Cells of Niabella species are Gram-stain-negative, aerobic, non-flagellated and short rods. Since the genus Niabella was firstly described by Kim et al., 11 specific names have been validly published. The names of ‘Niabella terrae’ and ‘Niabella thaonhiensis’ were proposed but have not been validated at the time of this writing. The genus Thermomonas, a member of the family Xanthomonadaceae, was firstly described by Busse et al.. At the time of writing, seven species have been validly published. The species of the genera Niabella and Thermomonas have been isolated from a wide range of habitats, and they share features such as cells that are Gram-strain-negative, aerobic and non-flagellated. In this report, we describe two bacterial strains that were isolated from a constructed wetland system, by phylogenetic, physiological, biochemical, and genomic analyses. The constructed wetland system, also called the Dragon-shaped Wetland, located in the central area of the Beijing Olympic Park area, is the largest urban artificial water system in Asia. The water in the Dragon-shaped Water System comes mainly from the Beijing Qinghe Water Reclamation Plant, and the entire water system has a complete circulation system. Sludge samples from the Dragon-shaped Wetland water system were collected, and were serially diluted with 0.85 % NaCl and plated onto Reasoner's 2A agar. The isolates, designated as 3A5MI-3T and RSS-23T, were obtained after incubation for 3 days at 30 °C and were routinely stored at −80 °C as suspensions in R2A broth supplemented with 20 % glycerol. Genomic DNA was extracted with a commercial TIANamp Bacteria DNA Kit, and 16S rRNA genes were amplified by PCR with universal bacterial primers 27 F and 1492R, which was also used for sequencing the PCR product. The almost-complete sequence was compared with 16S rRNA gene sequences from GenBank. The 16S rRNA gene sequences were aligned using ClustalW. Phylogenetic analyses were carried out using three phylogenetic algorithms: neighbour-joining, maximum-likelihood and maximum-parsimony. Phylogenetic trees were reconstructed and bootstrapped with 1000 replicates of each sequence using mega version 7.0. The CVTree method was used to reconstruct phylogenomic tree based on whole genomes. Analysis of 16S rRNA gene sequences in GenBank indicated that strain 3A5MI-3T was phylogenetically close to Niabella pedocola R384T, ‘Niabella thaonhiensis’ NHI-24T, Niabella aurantiaca R2A15-11T, Niabella tibetensis 15-4T, Niabella soli JS 13-8T, Niabella drilacis E90T, Niabella hirudinis CCM8411T, Niabella aquatica RP-2T, Niabella yanshanensis CCBAU 05354T, Niabella ginsenosidivorans BS26T, ‘Niabella terrae’ ICM 1-15T, Niabella hibiscisoli THG-DN5.5T and Niabella ginsengisoli GR 10-1T, as well as to Terrimonas rhizosphaerae CR94T and was &lt;92 % similar to some other species included in this analysis. Based on the results of phylogenetic analysis and 16S rRNA gene identity, N. pedocola JCM 31011T was selected as a reference strain for phenotypic tests. Strain RSS-23T was phylogenetically close to Thermomonas aquatica SY21T, Thermomonas fusca LMG 21737T, Thermomonas brevis LMG 21746T, Thermomonas carbonis GZ436T, Thermomonas haemolytica A50-7-3T, Thermomonas koreensis NBRC 101155T and Thermomonas hydrothermalis SGM-6T. The phylogenetic tree showed that the strain grouped with seven members of the genus Thermomonas. A phylogenomic tree based on genome sequences also showed the phylogenetic positions of strains 3A5MI-3T and RSS-23T. To analyse the genomic properties of strains 3A5MI-3T and RSS-23T, whole-genome sequencing was performed using the Illumina system. The genomic assembly was performed with the SPAdes software using clean data. Average nucleotide identity values were calculated with ChunLab’s online ANI Calculator. Digital DNA–DNA hybridization was performed by using the Genome-to-Genome Distance Calculator. The genome of strain 3A5MI-3T contained three contigs, and the total length was 6.59 Mb, encoding 6358 genes. The calculated DNA G+C content of strain 3A5MI-3T was 47.07 mol%. The genome of strain 3A5MI-3T carries 49 ncRNA genes, including three rRNA operons, 41 tRNA genes and five sRNA genes. In addition, there is a clustered regularly interspaced short palindromic repeat sequence with a length of 2862 bp in this genome. The Kyoto Encyclopedia of Genes and Genomes functional category distribution revealed that large numbers of genes in the genome of strain 3A5MI-3T belonged to the metabolism, genetic information processing, environmental information processing and cellular processes. Annotated with the RefSeq non-redundant proteins database, the genomes of strain 3A5MI-3T harboured nine genes encoding phosphohydrolase, nine genes encoding phosphatase, 11 genes encoding glucosidase, 51 genes encoding glycosyl hydrolase families, 36 genes encoding starch binding protein, six genes encoding polysaccharide lyase, seven genes encoding polysaccharide deacetylase, 28 genes encoding SusC/RagA family TonB-linked outer membrane protein, 13 genes encoding xylanase, one gene encoding cellulase and one gene encoding chondroitinase-B. We found genes for xylanase and cellulase in the strain 3A5MI-3T and N. pedocola JCM31011T genomes. APIZYM and carbon source utilization experiments confirmed that strain 3A5MI-3T and N. pedocola JCM31011T could utilize xylan and cellulose. Although strain 3A5MI-3T and N. pedocola JCM31011T both have the chitinase gene in their genome, neither of them can use chitin. Trypsin-like serine proteases were found in the strain 3A5MI-3T genome but not in strain N. pedocola JCM31011T, which corresponds to APIZYM trypsin activity for strain 3A5MI-3T but not for N. pedocola JCM31011T. Comparing the genomes of strain 3A5MI-3T and N. pedocola JCM 31011T, we found that the specific genes detected in strain 3A5MI-3T were as follows: four genes encoding lipopolysaccharide biosynthesis protein, three genes encoding biotin synthase, three genes encoding CRISPR-associated protein Cas1, two genes encoding 2-dehydropantoate 2-reductase, one gene encoding bifunctional NAD(P)H-nitrite reductase/anaerobic dehydrogenase, two genes encoding arabinogalactan endo-1,4-beta-galactosidase, three genes encoding TDP-4-oxo-6-deoxy-d-glucose aminotransferase, one gene encoding zinc-binding alcohol dehydrogenase family protein, three genes encoding 3-phytase, three genes encoding alpha-glucuronidase and one gene encoding Alg9-like mannosyltransferase family protein. The accession numbers of the sequences in the database relating to the specific or characteristic genes of strain 3A5MI-3T and N. pedocola JCM 31011T are provided as supplementary material. The ANI value between strains 3A5MI-3T and N. pedocola JCM 31011T was calculated using ChunLab’s online ANI Calculator. We performed genome sequencing of N. pedocola JCM31011T and its GenBank/EMBL/DDBJ accession number is JAJNEC000000000. The ANI result was 77.48 %, below the 95 % inter-species threshold. dDDH was performed with genome sequences of strains 3A5MI-3T and N. pedocola JCM 31011T using the Genome-to-Genome Distance Calculator 29]. The estimated dDDH value was 19.50 %, which was below the standard value generally recommended for species differentiation. The genome of strain RSS-23T contained also three contigs, and the total length is 2.79 Mb, encoding 2575 genes. The calculated genomic DNA G+C content of strain RSS-23T was 61.21 mol%. The genome of strain RSS-23T carries 58 ncRNA genes, including six rRNA genes, 49 tRNA genes and three sRNA genes. The KEGG functional category distribution revealed that large numbers of genes in the genome of strain RSS-23T belonged to the metabolism, genetic information processing, environmental information processing, cellular processes. Annotated with the RefSeq NR database, the genomes of strain RSS-23T harboured 61 genes encoding phosphatase, six genes encoding glucosidase, four genes encoding glycosyl hydrolase, 27 genes encoding glycosyl transferase and 11 genes encoding polysaccharide biosynthesis protein. APIZYM positive enzyme genes such as alkaline phosphatase, acid phosphatase and α-glucosidase were also found in the genome. Comparing the genomes of strains RSS-23T and T. fusca DSM 15424T, we found that the specific genes contained in strain RSS-23T are as follows: three genes encoding aminotransferase class I/II, six genes encoding chloride channel protein, three genes encoding penicillin acylase, three genes encoding UDP-glucose 4-epimerase, three genes encoding UDP-N-acetylglucosamine 2-epimerase and three genes encoding carbonate dehydratase. The accession numbers of the sequences on the database relating to the specific or characteristic genes of strain RSS-23T and T. fusca DSM 15424T are provided as supplementary files. The ANI values between strain RSS-23T and T. fusca DSM 15424T or T. aquatica SY21T were 78.10 and 76.11 %, respectively, which were below the 95 % inter-species threshold. The estimated dDDH values of strain RSS-23T to T. fusca DSM 15424T and T. aquatica SY21T were 25.20 and 20.90 %, respectively, which were below the standard value generally recommended for species differentiation. The morphology and size of cells grown on R2A agar at 30 °C for 2 days were observed by using transmission electron microscopy, and motility was observed with light microscopy. Gram-staining was performed according to Hucker. Growth was measured at temperatures of 16, 20, 30, 37,40, 45, 50, 60, 65 and 70 °C. Anaerobic growth was examined in R2A broth without oxygen, and using l-cysteine to consume residual oxygen and resazurin as a redox indicator. Salt tolerance was investigated by supplementing NaCl concentrations of 0, 1, 2, 3, 4 and 5.0 % to R2A broth. The effect of pH was tested in R2A broth using phosphate buffer and Tris buffer. Catalase activity was determined using a 3 % hydrogen peroxide solution. Oxidation of N,N,N′,N′-tetramethyl-p-phenylenediamine dihydrochloride was used to check oxidase activity. Monosaccharide, disaccharide and polysaccharide utilization was tested with the following compounds and basic inorganic salt at concentration of 1 g 1−1: starch, cellulose, chitin, xylan, agar, fructose, galactose, mannose, xylose, arabinose, glucose, lactose, maltose, cellobiose, trehalose, sucrose and sorbose. Bacterial cells grown in R2A were collected and washed three times with basic inorganic salt medium, inoculated at 1 %, and cultivated for 2 days. Other physiological and biochemical analyses were carried out by using the API ZYM, API 20NE and Biolog GEN III systems, according to the manufacturers’ instructions. Cells of strains 3A5MI-3T and RSS-23T were Gram-stain-negative, aerobic, non-motile and short rod-shaped. The distinctive physiological characteristics of strains 3A5MI-3T and RSS-23T are listed in Tables 1 and 2, respectively. The cellular fatty acids were determined using cells grown on R2A agar for 2 days at 30 °C with the standard midi protocol and analysed with a gas chromatograph. The extraction, purification and analysis of quinones were carried out according to the methods of Collins et al.. The total lipids were extracted using chloroform and methanol, followed by thin-plate biphasic chromatography to identify each lipid component. Strain 3A5MI-3T contained aliphatic saturated fatty acid as its predominant cellular fatty acid that accounted for about half of total fatty acids, followed by iso-C15 : 1 ω6c and/or iso-C15 : 1 ω7c, iso-C17 : 0 3-OH and summed feature 3. The composition of the fatty acids is shown in Table 3. The major cytoquinone was determined to be MK-7, as previously reported for described members of the genus Niabella. The polar lipid profile of strain 3A5MI-3T contained phosphatidylethanolamine and three unknown lipids. Strain RSS-23T also contained aliphatic saturated fatty acid as its predominant cellular fatty acid, which accounted for about one third of total fatty acids. The fatty acid profiles are shown in Table 4. The major isoprenoid quinone was determined to be Q-8, as previously reported for members of the genus Thermomonas. The polar lipid profile contained diphosphatidylglycerol, phosphatidylglycerol, phosphatidylethanolamine, two unknown phospholipids and an unknown lipid. Based on the phylogenetic and phenotypic characteristics, we conclude that strains 3A5MI-3T and RSS-23T represent novel species of the genera Niabella and Thermomonas, respectively, for which the names Niabella beijingensis sp. nov. and Thermomonas beijingensis sp. nov. are proposed. Niabella beijingensis. Cells are Gram-stain-negative, strictly aerobic, non-motile and short rod-shaped. Colonies are convex, circular, smooth and edges were regular after incubation on R2A broth for 2 days at 30 °C. The strain can grow in a wide range of temperature and pH, and the range of NaCl tolerance is 0–2 % NaCl. Optimal growth at 30 °C, pH 7.0 and with 1 % NaCl. Oxidase and catalase activities are positive and negative, respectively. According to the API 20NE system, nitrate cannot be reduced to nitrite. Positive for aesculin ferric citrate, gelatin, 4-nitrophenyl-β-d-galactopyranoside d-glucose, l-arabinose, d-mannose, N-acetyl-glucosamine and maltose, and negative for l-tryptophan, l-arginine, urea, d-mannitol, potassium gluconate, capric acid, adipic acid, malic acid, trisodium citrate and phenylacetic acid. The following enzyme activities are positive: alkaline phosphatase, leucine arylamidase, valine arylamidase, acid phosphatase, naphthol-AS-BI-phosphohydrolase, α-galactosidase, β-galactosidase, α-glucosidase, β-glucosidase, N-acetyl-β-glucosaminidase, α-mannosidase and α-fucosidase. Esterase lipase, cystine arylamidase and trypsin are weak; and esterase, lipase, α-chymotrypsin and β-glucuronidase are negative. Carbon substrates utilized include gentiobiose, sucrose, stachyose, maltose, trehalose, cellobiose, turanose, raffinose, lactose, melibiose, d-mannose, d-fructose, d-galactose, l-rhamnose, glycy-l-proline, l-aspartic acid, l-glutamic acid, l-serine, dextrin, pectin, Tween 40, acetoacetic acid, acetic acid, l-lactic acid, N-acetyl neuraminic acid, d-galacturonic acid, d-glucuronic acid, l-galactonic acid lactone, methyl β-d-glucoside, d-salicin, N-acetyl-d-glucosamine, N-acetyl-d-galactosamine, glucuronamide, gelatin, d-arabinose, d-sorbose, starch and agar. Weak utilization of l-fucose, d-glucose-6-PO4, l-alanine, l-histidine, d-gluconic acid, citric acid, glycerol, N-acetyl-β-d-mannosamine, methyl pyruvate, xylan and cellulose. No utilization of 3-methyl glucose, d-fucose, inosine, d-sorbitol, d-mannitol, d-arabitol, myo-inositol, d-fructose-6-PO4, d-aspartic acid, d-serine, l-arginine, l-pyroglutamic acid, mucic acid, quinic acid, d-saccharic acid, p-hydroxy-phenylacetic acid, d-lactic acid methyl ester, α-keto-glutaric acid, d-malic acid, l-malic acid, bromo-succinic acid, γ-amino-butryric acid, β-hydroxy-d,l-butyric acid, α-keto-butyric acid, propionic acid, formic acid and chitin. The predominant cellular fatty acids are iso-C15 : 0, iso-C15 : 1 ω6c and/or iso-C15 : 1 ω7c, iso-C17 : 0 3-OH and summed feature 3. Contains cytoquinone MK-7 as respiratory quinone. The polar lipid profile contains phosphatidylethanolamine and three unknown lipids. The G+C content of the type strain is 47.07 mol%. The type strain, 3A5MI-3T, was isolated from constructed wetland sludge from Beijing, PR China. Thermomonas beijingensis. Cells are Gram-stain-negative, aerobic, non-motile and rod-shaped. Colonies are convex, circular, smooth and edges are regular after incubation on R2A broth for 2 days at 30 °C. The strain can grow in a wide range of temperature and pH. The range of NaCl tolerance is 0–1 % NaCl. Optimal growth at 30 °C, pH 7.0 and with 1 % NaCl. Oxidase and catalase activities are positive. According to the API 20NE system, nitrate can be reduced to nitrite. Positive for aesculin ferric citrate, gelatine, d-glucose, N-acetyl-glucosamine and maltose, and negative for l-tryptophan, l-arginine, urea, 4-nitrophenyl-β-d-galactopyranoside, l-arabinose, d-mannose, d-mannitol, potassium gluconate, caprate, adipate, malate, citrate and phenylacetate. The following enzyme activities of alkaline phosphatase, leucine arylamidase, valine arylamidase, acid phosphatase, naphthol-AS-BI-phosphohydrolase, α-glucosidase, β-glucosidase, N-acetyl-β-glucosaminidase, esterase lipase, cystine arylamidase, trypsin, esterase and α-chymotrypsin are positive; α-mannosidase. α-fucosidase, lipase, α-galactosidase, β-galactosidase and β-glucuronidase are negative. Carbon sources utilized include dextrin, maltose, cellobiose, gentiobiose, N-acetyl-d-glucosamine, N-acetyl-d-galactosamine, gelatin, glycy-l-proline, l-alanine, l-arginine, l-aspartic acid, l-glutamic acid, l-serine, methyl pyruvate, Tween 40, α-hydroxy-butyric acid, β-hydroxy-d,l-butyric acid, α-keto-butyric acid, acetoacetic acid, propionic acid, acetic acid, d-arabinose, d-sorbose, starch, xylan and agar. Weak utilization of turanose, N-acetyl-β-d-mannosamine, d-fructose, d-glucose-6-PO4, d-aspartic acid and pectin. No growth occurs on trehalose, sucrose, stachyose, raffinose, lactose, melibiose, methyl β-d-glucoside, d-salicin, N-acetyl neuraminic acid, α-d-glucose, d-mannose, d-galactose, 3-methyl glucose, d-fucose, l-fucose, l-rhamnose, inosine, d-sorbitol, d-mannitol, d-arabitol, myo-inositol, glycerol, d-serine, l-histidine, l-pyroglutamic acid, d-galacturonic acid, l-galactonic acid lactone, d-gluconic acid, d-glucuronic acid, glucuronamide, mucic acid, quinic acid, d-saccharic acid, p-hydroxy-phenylacetic acid, d-lactic acid methyl ester, l-lactic acid, citric acid, α-keto-glutaric acid, d-malic acid, l-malic acid, bromo-succinic acid, γ-amino-butryric acid, formic acid, cellulose and chitin. The predominant cellular fatty acids are iso-C15 : 0, iso-C17 : 1 ω9c, iso-C11 : 0 3-OH and summed feature 3. The major isoprenoid quinone is Q-8. The polar lipid profile contains diphosphatidylglycerol, phosphatidylglycerol, phosphatidylethanolamine, two unknown phospholipids and an unknown lipid. The type strain, RSS-23T, was isolated from soil of the Dragon-shaped Wetland System in Beijing Olympic Park, PR China. The DNA G+C content of the type strain is 61.21 mol%. This work was financially supported by National Natural Science Foundation of China. We thank Prof. Yuguang Zhou at Institute of Microbiology, Chinese Academy of Sciences for coordination of deposits of type strains. The authors declare that there are no conflicts of interest. Phenotypic characteristics that differentiate strain 3A5MI-3T from the phylogenetically closely related strains of the genus NiabellaStrains: 1, 3A5MI-3T; 2, N. pedocola JCM 31011T; 3, N. aquatica JCM 30952T; 4, N. aurantiaca DSM 17617T; 5, N. ginsengisoli JCM 15444T; 6, N. ginsenosidivorans JCM 18199T; 7, N. hibiscisoli KACC 18857T; 8, N. hirudinis DSM 25812T; 9, N. soli DSM 19437T; 10, ‘N. terrae’ JCM 19502T; 11, ‘N. thaonhiensis’ JCM 18864T; 12, N. yanshanensis KACC 14980T; 13, N. drilacis DSM25812T; 14, N. tibetensis CCTCC AB 209167T. +, Positive; −, negative; w, weakly positive. The data of strain 3A5MI-3T and N. pedocola JCM 31011T were taken from the current study, others were taken from the published literature. Phenotypic characteristics that differentiate strain RSS-23T from the phylogenetically closely related species of the genus ThermomonasStrains: 1, RSS-23T; 2, T. fusca DSM 15424T; 3, T. aquatica KCTC 62191T; 4, T. carbonis KCTC 42013T; 5, T. brevis DSM 15422T; 6, T. haemolytica DSM 13605T; 7, T. koreensis KCTC 12540T; 8, T. hydrothermalis DSM 14834T. +, Positive; −, negative; w, weakly positive/sensitive. The data of strain RSS-23T and T. fusca DSM 15424T were taken from the current study, others were taken from the published literature. Cellular fatty acid composition of strain 3A5MI-3T and closely related species of the genus NiabellaStrains: 1, 3A5MI-3T; 2, N. pedocola JCM 31011T; 3, N. aquatica JCM 30952T; 4, N. aurantiaca DSM 17617T; 5, N. ginsengisoli JCM 15444T; 6, N. ginsenosidivorans JCM 18199T; 7, N. hibiscisoli KACC 18857T; 8, N. hirudinis DSM 25812T; 9, N. soli DSM 19437T; 10, ‘N. terrae’ JCM 19502T; 11, ‘N. thaonhiensis’ JCM 18864T; 12, N. yanshanensis KACC 14980T ; 13，N. drilacis DSM25812T; 14, N. tibetensis CCTCC AB 209167T. −, Not detected or &lt;0.1 %. The data of strain 3A5MI-3T and N. pedocola JCM 31011T were taken from the current study, others were taken from the published literature. Cellular fatty acid composition of strain RSS-23T and closely related species of the genus ThermomonasStrains: 1, RSS-23T; 2, T. fusca DSM 15424T; 3, T. aquatica KCTC 62191T; 4, T. carbonis KCTC 42013T; 5, T. brevis DSM 15422T; 6, T. haemolytica DSM 13605T; 7, T. koreensis KCTC 12540T; 8, T. hydrothermalis DSM 14834T. −, Not detected or &lt;0.1 %. The data of strain RSS-23T and T. fusca DSM 15424T were taken from the current study, others were taken from the published literature."
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6b8b061a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cluster 1\n",
      "this work\n",
      "the article\n",
      "this report\n",
      "\n",
      "Cluster 2\n",
      "Bioproject\n",
      "Reasoner's\n",
      "\n",
      "Cluster 3\n",
      "members of the genera Niabella and Thermomonas\n",
      "they\n",
      "\n",
      "Cluster 4\n",
      "the genera Niabella\n",
      "The genus Niabella\n",
      "the genus Niabella\n",
      "\n",
      "Cluster 5\n",
      "the genera Niabella and Thermomonas\n",
      "the genera Niabella and Thermomonas\n",
      "\n",
      "Cluster 6\n",
      "Thermomonas\n",
      "Thermomonas\n",
      "\n",
      "Cluster 7\n",
      "seven species\n",
      "The species of the genera Niabella and Thermomonas\n",
      "they\n",
      "\n",
      "Cluster 8\n",
      "a constructed wetland system\n",
      "The constructed wetland system, also called the Dragon-shaped Wetland, located in the central area of the Beijing Olympic Park area\n",
      "the Dragon-shaped Water System\n",
      "the entire water system\n",
      "the Dragon-shaped Wetland water system\n",
      "\n",
      "Cluster 9\n",
      "a constructed wetland system\n",
      "The constructed wetland system, also called the Dragon-shaped Wetland, located in the central area of the Beijing Olympic Park area\n",
      "the entire water system\n",
      "the Dragon-shaped Wetland water system\n",
      "\n",
      "Cluster 10\n",
      "PCR with universal bacterial primers 27 F and 1492R, which was also used for sequencing the PCR product\n",
      "PCR\n",
      "\n",
      "Cluster 11\n",
      "GenBank\n",
      "GenBank\n",
      "\n",
      "Cluster 12\n",
      "Strain RSS-23T\n",
      "the strain\n",
      "\n",
      "Cluster 13\n",
      "strains 3A5MI-3T\n",
      "strains 3A5MI-3T\n",
      "\n",
      "Cluster 14\n",
      "strains 3A5MI-3T and RSS-23T\n",
      "strain 3A5MI-3T and N. pedocola JCM31011T\n",
      "strain 3A5MI-3T and N. pedocola JCM31011T\n",
      "their\n",
      "them\n",
      "\n",
      "Cluster 15\n",
      "strains 3A5MI-3T and RSS-23T\n",
      "strain 3A5MI-3T and N. pedocola JCM31011T\n",
      "strain 3A5MI-3T and N. pedocola JCM31011T\n",
      "their\n",
      "them\n",
      "strain 3A5MI-3T and N. pedocola JCM 31011T\n",
      "strain 3A5MI-3T and N. pedocola JCM 31011T\n",
      "strains 3A5MI-3T and N. pedocola JCM 31011T\n",
      "\n",
      "Cluster 16\n",
      "RSS-23T\n",
      "N. pedocola JCM31011T\n",
      "N. pedocola JCM31011T\n",
      "N. pedocola JCM31011T\n",
      "\n",
      "Cluster 17\n",
      "RSS-23T\n",
      "N. pedocola JCM31011T\n",
      "N. pedocola JCM31011T\n",
      "N. pedocola JCM31011T\n",
      "strain N. pedocola JCM31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM31011T\n",
      "its\n",
      "N. pedocola JCM 31011T\n",
      "\n",
      "Cluster 18\n",
      "The genome of strain 3A5MI-3T\n",
      "The genome of strain 3A5MI-3T\n",
      "this genome\n",
      "the genome of strain 3A5MI-3T\n",
      "the genomes of strain 3A5MI-3T\n",
      "\n",
      "Cluster 19\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "\n",
      "Cluster 20\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strains 3A5MI-3T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "\n",
      "Cluster 21\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strains 3A5MI-3T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "\n",
      "Cluster 22\n",
      "APIZYM\n",
      "APIZYM\n",
      "\n",
      "Cluster 23\n",
      "we\n",
      "We\n",
      "\n",
      "Cluster 24\n",
      "The ANI value between strains 3A5MI-3T and N. pedocola JCM 31011T\n",
      "The ANI result\n",
      "\n",
      "Cluster 25\n",
      "ANI\n",
      "ANI\n",
      "\n",
      "Cluster 26\n",
      "The genome of strain RSS-23T\n",
      "The genome of strain RSS-23T\n",
      "\n",
      "Cluster 27\n",
      "The genome of strain RSS-23T\n",
      "The genome of strain RSS-23T\n",
      "the genome of strain RSS-23T\n",
      "the genomes of strain RSS-23T\n",
      "the genome\n",
      "\n",
      "Cluster 28\n",
      "the RefSeq NR database\n",
      "the database\n",
      "\n",
      "Cluster 29\n",
      "strains RSS-23T and T. fusca DSM 15424T\n",
      "strain RSS-23T and T. fusca DSM 15424T\n",
      "strain RSS-23T and T. fusca DSM 15424T\n",
      "\n",
      "Cluster 30\n",
      "T. fusca DSM 15424T\n",
      "T. fusca DSM 15424T\n",
      "T. fusca DSM 15424T\n",
      "T. fusca DSM 15424T\n",
      "\n",
      "Cluster 31\n",
      "DSM\n",
      "DSM\n",
      "\n",
      "Cluster 32\n",
      "R2A\n",
      "R2A\n",
      "\n",
      "Cluster 33\n",
      "R2A broth\n",
      "R2A broth\n",
      "\n",
      "Cluster 34\n",
      "strains 3A5MI-3T\n",
      "strains 3A5MI-3T\n",
      "\n",
      "Cluster 35\n",
      "strains 3A5MI-3T and RSS-23T\n",
      "strains 3A5MI-3T and RSS-23T\n",
      "\n",
      "Cluster 36\n",
      "RSS-23T\n",
      "RSS-23T\n",
      "\n",
      "Cluster 37\n",
      "Strain 3A5MI-3T\n",
      "its\n",
      "strain 3A5MI-3T\n",
      "\n",
      "Cluster 38\n",
      "Strain RSS-23T\n",
      "its\n",
      "\n",
      "Cluster 39\n",
      "strains 3A5MI-3T and RSS-23T\n",
      "The strain\n",
      "\n",
      "Cluster 40\n",
      "l-fucose\n",
      "d-fucose\n",
      "\n",
      "Cluster 41\n",
      "d-glucose-6-PO4\n",
      "d-fructose-6-PO4\n",
      "\n",
      "Cluster 42\n",
      "the type strain\n",
      "The type strain, 3A5MI-3T,\n",
      "The strain\n",
      "\n",
      "Cluster 43\n",
      "N-acetyl-glucosamine\n",
      "N-acetyl-d-glucosamine\n",
      "\n",
      "Cluster 44\n",
      "The type strain, RSS-23T\n",
      "The type strain, RSS-23T,\n",
      "the type strain\n",
      "\n",
      "Cluster 45\n",
      "The type strain, RSS-23T\n",
      "the type strain\n",
      "\n",
      "Cluster 46\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "strain 3A5MI-3T\n",
      "strain 3A5MI-3T\n",
      "\n",
      "Cluster 47\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "\n",
      "Cluster 48\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "N. pedocola JCM 31011T\n",
      "T. fusca DSM 15424T\n",
      "\n",
      "Cluster 49\n",
      "KACC\n",
      "KACC\n",
      "\n",
      "Cluster 50\n",
      "strain 3A5MI-3T and N. pedocola JCM 31011T\n",
      "strain 3A5MI-3T and N. pedocola JCM 31011T\n",
      "\n",
      "Cluster 51\n",
      "strain 3A5MI-3T and N. pedocola JCM 31011T\n",
      "strain 3A5MI-3T and N. pedocola JCM 31011T\n",
      "strain RSS-23T and T. fusca DSM 15424T\n",
      "\n",
      "Cluster 52\n",
      "the current study\n",
      "the current study\n",
      "the current study\n",
      "\n",
      "Cluster 53\n",
      "JCM\n",
      "JCM\n",
      "\n",
      "Cluster 54\n",
      "strain RSS-23T\n",
      "strain RSS-23T\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# sorts coreferences by clusters\n",
    "i = 1\n",
    "for item in doc._.coref_clusters:\n",
    "    print (f'Cluster {i}')\n",
    "    for span in item:\n",
    "        start, end = span\n",
    "        print(doc[start:end+1])\n",
    "    i = i+1\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89c462f3",
   "metadata": {},
   "source": [
    "## Result\n",
    "crosslingual_coreference is not a good solution when it comes to scientific articles.\n",
    "The model has to be trained on the specific domain and is weak with longer sentences/texts. Proofed above. Acc about 0,8.\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "test",
   "language": "python",
   "name": "test"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
